### ISSUE TITLE
Pager App incorrect decoding and displaying of DAPNET messages (with possible buffer overflow)

### ISSUE THREAD

**Issue by dl9sec** [xMasterX/flipper-pager#14]
Hi,

unfortunately the App is totally unstable on my Flipper with RogueMaster (RM0429-1240-0.101.3-a5550b3).
After receiving several messages, the App completely freezes, sometimes also when viewing a message.
Leaving the App with the back button is sometimes impossible after some messages received.
The only way to kill the App is to reboot (left+back).
When bigger messages are viewed, the display is garbled (seems, that some of the text is written somewhere on the display).

Regards, Thorsten

**Comment by xMasterX** [comment]
Looks like your firmware is totally unstable, how about to test it on official firmware for example?
Latest .fap here: https://github.com/xMasterX/all-the-plugins/releases

![Screenshot-20240503-130038](https://github.com/xMasterX/flipper-pager/assets/10697207/d73471b7-365f-4033-bc4b-0bf770b12f3a)

**Comment by dl9sec** [comment]
Yes, I tested also with the official firmware and the "latest .fap". Same there.
As soon, as the second message was received the blinking of the LED gets slower. When pressing back button, LED goes off and nothing happens anymore (waited several minutes).
Is there a reason why the "Latest .fap" shows "Version: 0.1" instead of "Version: 1.2"?

**Comment by xMasterX** [comment]
> "Version: 1.2"?

This will be fixed

I was unable to replicate anything described, can you provide raw signals that are causing such issues?
Or their full unedited text data?

**Comment by dl9sec** [comment]
> can you provide raw signals that are causing such issues? Or their full unedited text data?

How? I can not choose DAPNET frequency 439.9875 in the Sub-GHz app...
Edit: Oh, I saw that there is a possibility to edit frequencies in /subghz/assets/settings_user... Will try that

I also tried the latest UnleashedFW 072. Same there.
For me it seems, that transmissions are mixed up in the decoder:

![Screenshot-20240503-143055](https://github.com/xMasterX/flipper-pager/assets/7019835/5205a9c1-3e1a-4757-9f0d-6bf5d1100890)

It seems, that exactly those three messages are mixed up in one receive entry:

![2024-05-03 14_31_30-DL9SEC - Digital Voice Tableau – Mozilla Firefox](https://github.com/xMasterX/flipper-pager/assets/7019835/5d5e2249-78aa-4c94-8b87-afbc21feaf6c)

Maybe this causes an overflow somewhere...

**Comment by dl9sec** [comment]
Here is some RAW data. These messages (2..4 in sequence) seems not to be separated correctly by the decoder. After getting such message sequences, the LED blinking frequency decreases and the keypad input seems to be delayed extremely...
[Raw-data_examples.zip](https://github.com/xMasterX/flipper-pager/files/15201031/Raw-data_examples.zip)

**Comment by dl9sec** [comment]
The Sub-GHz app seems to behave more better to the sequence messages. They are also mixed up into one message with the RIC of the last received message, but the LED blink frequency doesn't change and the buttons are always responsive....

**Comment by dl9sec** [comment]
Thank you for caring about that issue. Let me know, if  you need further information.
Regards...

**Comment by xMasterX** [comment]
I will try to replicate the issue using your captures, thanks for sharing

**Comment by dl9sec** [comment]
It could be, that the Pi-Star DAPNET gateway is responsible for the decoding problem. Yesterday I got an Alphapoc 602R pager and it has a similar problem decoding such multiple message transmissions (see https://forum.pistar.uk/viewtopic.php?t=4921).
Will see, if some of the Pi-Star guys has an idea...

**Comment by dl9sec** [comment]
See also https://github.com/g4klx/DAPNETGateway/issues/35
